internal/gatewayapi/route.go

```diff
@@ -486,53 +484,37 @@ func (t *Translator) processHTTPRouteParentRefListener(route RouteContext, route
 		}
 		hasHostnameIntersection = true
 
+		// only represent non-trivial hosts in ir
+		if len(hosts) == 1 && hosts[0] == "*" {
+			hosts = nil
+		}
+
 		var perHostRoutes []*ir.HTTPRoute
-		for _, host := range hosts {
-			var headerMatches []*ir.StringMatch
-
-			// If the intersecting host is more specific than the Listener's hostname,
-			// add an additional header match to all of the routes for it
-			if host != "*" && (listener.Hostname == nil || string(*listener.Hostname) != host) {
-				// Hostnames that are prefixed with a wildcard label (*.)
-				// are interpreted as a suffix match.
-				if strings.HasPrefix(host, "*.") {
-					headerMatches = append(headerMatches, &ir.StringMatch{
-						Name:   ":authority",
-						Suffix: StringPtr(host[2:]),
-					})
-				} else {
-					headerMatches = append(headerMatches, &ir.StringMatch{
-						Name:  ":authority",
-						Exact: StringPtr(host),
-					})
-				}
+		for _, routeRoute := range routeRoutes {
```

Comment:
looks like `perHostRoutes` is no longer needed, if thats the case, will it be simpler if the code looked lik
```
for _, routeRoute := range routeRoutes {
    routeRoute.Hostnames = hosts
```